src/CLICommand/commonContext.ts

```diff
@@ -0,0 +1,89 @@
+import { DriveKey } from '../arfsdao';
+import { JWKWallet, Wallet, WalletDAO } from '../wallet_new';
+import { ParameterName } from './parameter';
+import * as fs from 'fs';
+import { JWKInterface } from 'ardrive-core-js';
+import {
+	DriveAddressParameter,
+	DriveKeyParameter,
+	DrivePasswordParameter,
+	SeedPhraseParameter,
+	WalletFileParameter
+} from '../parameter_declarations';
+import Arweave from 'arweave';
+
+/**
+ * @type {CommonContext}
+ * A class representing the context of the parameters
+ */
+export class CommonContext {
+	private readonly walletDao: WalletDAO;
+
+	/**
+	 * @returns {CommonContext}
+	 * @param {any} options The object containing the parameterName: value mapping
+	 * @param {Arweave} arweave The arweave instance
+	 * An immutable instance of CommonContext holding the parsed values of the parameters
+	 */
+	constructor(private readonly options: any, private readonly arweave: Arweave) {
+		this.walletDao = new WalletDAO(this.arweave);
+	}
+
+	/**
+	 * @returns {Promise<boolean>}
+	 * Returns true when a drive password, drive key or wallet file is provided
+	 */
+	public async getIsPrivate(): Promise<boolean> {
+		return !!(this.password || this.driveKey || (await this.getWallet().catch(() => false)));
+	}
+
+	/**
+	 * @returns {Promise<Wallet>}
+	 * Will return a wallet instance created from the seed phrase or the walletFile
+	 */
+	public async getWallet(): Promise<Wallet> {
+		const walletFile = this.getParameterValue(WalletFileParameter);
```

Comment:
I'm trying to test the `get-balance` command with this setup and found that `walletFile` here console logs out as `true` rather than a string file path

Reply:
CLICommand was assuming all commands were boolean. Now you can specify the desired type, which defaults to 'single-value'. Fixed at ecb1cbc